### PR TITLE
CO: Specific price for Zone

### PR DESCRIPTION
**Pull request by christianverardi**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | adding specific price for Zone
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | [BOOM-2160](http://forge.prestashop.com/browse/BOOM-2160)
| How to test?  | 1. Clear the Symfony cache in order to recompile the YAML files
||2. Run the query to add a new field in the ps_specific_price table: `ALTER TABLE  PREFIX_specific_price ADD  id_zone int(10) unsigned NOT NULL AFTER id_country;`
||3. Run ALTER TABLE `PREFIX_specific_price` DROP INDEX `id_product_2`,
||4. ADD UNIQUE INDEX `id_product_2` (`id_product`,`id_product_attribute`,`id_customer`,`id_cart`,`from`,`to`,`id_shop`,`id_shop_group`,`id_currency`,`id_country`,`id_zone`,`id_group`,`from_quantity`,`id_specific_price_rule`);
||5. Run ALTER TABLE `PREFIX_specific_price` DROP INDEX `id_product`,
||6. ADD INDEX `id_product` (`id_product`,`id_shop`,`id_currency`,`id_country`,`id_zone`,`id_group`,`id_customer`,`from_quantity`,`from`,`to`);
||7. Create a new specific price for a product assigning a zone.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8316)

### CODE REVIEW

File: classes/SpecificPrice.php
Comment (LittleBigDev) [inline classes/SpecificPrice.php:217]: `$id_zone` parameter is not used in this method

File: classes/SpecificPrice.php
Comment (LittleBigDev) [inline classes/SpecificPrice.php:37]: Is snake_case mandatory here ? If not, please use camelCase.

**Review by LittleBigDev — CHANGES_REQUESTED** [review]
Hello Christian,
Thanks for this PR !

Few points to clarify before approval though.

File: classes/SpecificPrice.php
Comment (christianverardi) [reply classes/SpecificPrice.php:37]: I must leave snake_case because otherwise I would change the table too

File: classes/SpecificPrice.php
Comment (christianverardi) [reply classes/SpecificPrice.php:217]: it is called dynamically

**Review by LittleBigDev — DISMISSED** [review]
Approved after clarification.

**Comment by christianverardi** [comment]
@LittleBigDev I think that the tests have some problems. I saw the same errors in other PR's tests

**Comment by LittleBigDev** [comment]
@Quetzacoalt91 is currently working on it. Let's wait for his feedback.

**Comment by christianverardi** [comment]
@LittleBigDev is it possible to run the tests again?

**Comment by vincentbz** [comment]
Sure ! We review/test it soon, thanks :)

**Review by LittleBigDev — APPROVED** [review]
Travis tests went ok this time.
Codacy tests, well.. this is legacy. Some issues are impossible to fix without heavy refactoring. This is beyond this PR's scope.
Some comments are relevant, like "avoid adding more parameters to methods" but I guess this is too late to ask you to find another solution.
PrestaShop team has to settle this and set strict rules about legacy. You'll hear about it soon.
For now, this PR is ok(ish). Maybe @Quetzacoalt91, @eternoendless or @jocel1 can have a look at it too.

**Comment by vincentbz** [comment]
Hi @christianverardi ,

If I try to set a specific price on a zone, I have an error "undefined". If I don't select a zone it's working.
I've installed a new shop on a new database

image

**Comment by marionf** [comment]
Hello @christianverardi 
I have the same problem as @vincentbz

**Comment by marionf** [comment]
Hey @christianverardi 
Without answer from you, we are closing this PR.
Fell free to open a new one.
Thanks !

**Pull request closed without merging**
